src/NosCore.Controllers/GroupPacketController.cs

```diff
@@ -0,0 +1,330 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using NosCore.Configuration;
+using NosCore.GameObject;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.Networking;
+using NosCore.Packets.ClientPackets;
+using NosCore.Packets.ServerPackets;
+using NosCore.Shared.Enumerations;
+using NosCore.Shared.Enumerations.Character;
+using NosCore.Shared.Enumerations.Group;
+using NosCore.Shared.Enumerations.Interaction;
+using NosCore.Shared.I18N;
+
+namespace NosCore.Controllers
+{
+    public class GroupPacketController : PacketController
+    {
+        private readonly WorldConfiguration _worldConfiguration;
+
+        [UsedImplicitly]
+        public GroupPacketController()
+        {
+
+        }
+
+        public GroupPacketController(WorldConfiguration worldConfiguration)
+        {
+            _worldConfiguration = worldConfiguration;
+        }
+
+        /// <summary>
+        ///     pjoin packet
+        /// </summary>
+        /// <param name="pjoinPacket"></param>
+        public void ManageGroup(PjoinPacket pjoinPacket)
+        {
+            var targetSession = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == pjoinPacket.CharacterId);
+
+            if (targetSession == null && pjoinPacket.RequestType != GroupRequestType.Sharing)
+            {
+                return;
+            }
+
+            switch (pjoinPacket.RequestType)
+            {
+
+                case GroupRequestType.Requested:
+                case GroupRequestType.Invited:
+                    if (targetSession == null || pjoinPacket.CharacterId == Session.Character.CharacterId)
+                    {
+                        return;
+                    }
+
+                    if (targetSession.Character.Group?.IsGroupFull == true)
```

Comment:
Group should be empty and never null

Reply:
== false is usefull to avoid miss reading a ! but for true you can just (condition)